examples/rpc/partner-plugin/handlers/accommodation/v1/accommodation_search.go

```diff
@@ -0,0 +1,271 @@
+	available_properties := []*accommodationv1.PropertyExtendedInfo{}
+	// loop request queries
+	for _, query := range req.Queries {
+		props := make([]*accommodationv1.PropertyExtendedInfo, len(properties))
+		for i := range properties {
+			props[i] = &properties[i]
+		}
+
+		// get filtered properties
+		var filtered_props = filterPropertiesByGeoTreeLocation(props, query.SearchParametersAccommodation.GetLocationGeoTree())
+		// filter by product codes
+		filtered_props = filterPropertiesByProductCodes(filtered_props, query.SearchParametersAccommodation.GetProductCodes())
+		// filter by supplier codes
+		filtered_props = filterPropertiesBySupplierCodes(filtered_props, query.SearchParametersAccommodation.GetSupplierCodes())
+
+		// loop filtered properties and check if they are already in available_properties
+		for _, prop := range filtered_props {
+			// Check if property already exists in available_properties
+			exists := false
+			for _, existingProp := range available_properties {
+				if existingProp.Property.SupplierCode.SupplierCode == prop.Property.SupplierCode.SupplierCode {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				available_properties = append(available_properties, prop)
+			}
+		}
+
+		var price = 2
+
+		// generate search result
+		for _, prop := range available_properties {
+
+			// units requested
+			units_requested := query.UnitCount
+
+			// empty units array
+			units := make([]*accommodationv1.Unit, 0)
+
+			// loop all rooms
+			for _, room := range prop.Rooms {
+
+				units = append(units, &accommodationv1.Unit{
+					Type:             0,
+					SupplierRoomCode: room.SupplierCode,
+					SupplierRoomName: room.SupplierName,
+					OriginalRoomName: room.OriginalName,
+					TravelPeriod: &typesv1.TravelPeriod{
+						StartDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetStartDate().GetYear(),
+							Month: query.TravelPeriod.GetStartDate().GetMonth(),
+							Day:   query.TravelPeriod.GetStartDate().GetDay(),
+						},
+						EndDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetEndDate().GetYear(),
+							Month: query.TravelPeriod.GetEndDate().GetMonth(),
+							Day:   query.TravelPeriod.GetEndDate().GetDay(),
+						},
+					},
+					TravellerIds: getTravellerIds(query.Travellers),
+					Beds:         room.Beds,
+					PriceDetail: &typesv1.PriceDetail{
+						Price: &typesv1.Price{
+							Value: fmt.Sprintf("%d", price),
+							Currency: &typesv1.Currency{
+								Currency: req.SearchParametersGeneric.Currency.Currency,
+							},
+						},
+					},
+					Services:       []*typesv1.ServiceFact{},
+					MealPlanCode:   &typesv1.MealPlan{},
+					RatePlan:       &typesv1.RatePlan{},
+					RateRule:       &typesv1.RateRule{},
+					CancelPolicies: []*typesv1.CancelPolicy{},
+					RemainingUnits: 0,
+					PropertyCode:   &typesv1.ProductCode{},
+					SupplierCode:   prop.Property.SupplierCode,
+					Remarks:        "",
+				})
+
+				price += 2
```

Comment:
Modifying or overwriting the price of static mock data seems somewhat weird. 
Is there a reason why we're not defining the prices in the mock data and just return them?

Especially in the context that there is only one validation example which defines a static search id. Returning different search results for the same search id (or subsequentially verifying different search results with the same validation id) seems wrong.